src/functions/create-curation/handler.ts

```diff
@@ -25,22 +25,19 @@ const createCuration: ValidatedEventAPIGatewayProxyEvent<typeof schema> = async
     hidden,
     startTime,
     endTime,
-    title,
-    description,
-    links,
-    language,
+    document,
     curationType
   } = body;
   const { Authorization, authorization } = headers;
   const authHeader = Authorization || authorization;
 
-  const hasDocumentAttributes = !!(title && description && links && language);
-  if (curationType === CurationType.CUSTOM && !hasDocumentAttributes) {
+  if (curationType === CurationType.CUSTOM && !document) {
     return {
       statusCode: 400,
       body: "Bad request, missing document values"
     };
   }
+  const { title, description, links, language } = document!;
```

Comment:
These should be accessed only in the parts where they are required for the document related API calls. It also removes the need to use the non-null assertion, as we separately check that the document exists when we make those API calls.